kernels/ZendEngine3/fcall.c

```diff
@@ -420,7 +420,6 @@ int zephir_call_user_function(zval *object_pp, zend_class_entry *obj_ce, zephir_
 	fci.params         = NULL;
 	//fci.params: Passed as separate parameter to prevent the need to convert zval ** to zval *
 	fci.no_separation  = 1;
-	fci.symbol_table   = NULL;
```

Comment:
Because, in PHP 7.0.X, it's deprecated

    /* PHP-7 doesn't support symbol_table substitution for functions */
    ZEND_ASSERT(fci->symbol_table == NULL);

and in PHP 7.1.X (see master of PHP, it's deleted), see :

zend_API.h :

typedef struct _zend_fcall_info {
    size_t size;
    zval function_name;
    zval *retval;
    zval *params;
    zend_object *object;
    zend_bool no_separation;
    uint32_t param_count;
} zend_fcall_info;

so, it's not necessary to use it.

I saw in Zephir, no impact to delete this line and to set call->symbol_table
= NULL in "
extended/fcall.c".

2016-05-05 12:43 GMT+02:00 Steffen Butzer <notifications@github.com>:

> In kernels/ZendEngine3/fcall.c
> <https://github.com/phalcon/zephir/pull/1227#discussion_r62171992>:
>
> > @@ -420,7 +420,6 @@ int zephir_call_user_function(zval *object_pp, zend_class_entry *obj_ce, zephir_
> >  	fci.params         = NULL;
> >  	//fci.params: Passed as separate parameter to prevent the need to convert zval ** to zval *
> >  	fci.no_separation  = 1;
> > -	fci.symbol_table   = NULL;
>
> is there any reason you removed this line?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/phalcon/zephir/pull/1227/files/c2d15c599eeeff207834c732d7c7cd6a79d2331a#r62171992>
>